### PR TITLE
Amend Transport Permit Framework

### PR DESCRIPTION
**Pull request by dimak1**
*Issue #:* 
- bcgov/entity#19018

*Description of changes:*
- pre-fill Transport Permit for Amendment
- text updates as per designs
- fee summary
- unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/composables/mhrInformation/useMhrInfoValidation.ts [hunk 1]
```diff
@@ -85,13 +89,13 @@ export const useMhrInfoValidation = (validationState: mhrInfoValidationStateIF)
     if (isSameParkLocation) {
       return (
         (isRoleStaffReg.value ? validationState.isDocumentIdValid : true) &&
-        validationState.isLocationChangeTypeValid &&
+        (isAmendLocationActive.value ? true : validationState.isLocationChangeTypeValid) &&
```
Comment (dimak1) [inline ppr-ui/src/composables/mhrInformation/useMhrInfoValidation.ts:93]: Location Change dropdown does not exist for Amendments

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1741-cpkfb4pq.web.app

**Review by doug-lovett — APPROVED** [review]
Looks great.

**Review by cameron-eyds — COMMENTED** [review]
Looks good!

Imo it makes sense to include the snapshot parsing/storing/getter into the framework, as that will help set you up for the following component tickets but your call!

**Review by cameron-eyds — APPROVED** [review]

**Comment by dimak1** [comment]
@cameron-eyds do you mean parsing the existing Transport Permit data into the components for Amend? That's done. Or do you mean setter/getter of the original data for comparison with amended values?

**Comment by cameron-eyds** [comment]
> @cameron-eyds do you mean parsing the existing Transport Permit data into the components for Amend? That's done. Or do you mean setter/getter of the original data for comparison with amended values?

You bet, the latter. But your call, just usually part of the frameworks but no biggie.

File: ppr-ui/src/composables/mhrInformation/useTransportPermits.ts [hunk 2]
```diff
@@ -147,6 +160,21 @@ export const useTransportPermits = () => {
...
   }
 
+  // Pre-fill Transport Permit for Amendment
+  const prefillTransportPermit = () => {
+
+    const homeLocationInfo: MhrRegistrationHomeLocationIF  = getMhrRegistrationLocation.value
+    const ownLand = getMhrInformation.value.permitLandStatusConfirmation
+
+    // Set original Transport Permit for future comparison with Amendment filing
+    setMhrOriginalTransportPermit({ key: 'newLocation', value: homeLocationInfo })
```
Comment (dimak1) [inline ppr-ui/src/composables/mhrInformation/useTransportPermits.ts:170]: @cameron-eyds added setter for Original Transport Permit

**Comment by dimak1** [comment]
@cameron-eyds, you are right, the setter/getter should be part of this ticket. I added it here, thanks!

File: ppr-ui/src/composables/mhrInformation/useTransportPermits.ts (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/composables/mhrInformation/useTransportPermits.ts:170]: Beauty, nice work!

**Pull request merged**